### DISCUSSION TITLE
Can we render OJS or Julia output in PDFs?

### DISCUSSION THREAD

**Discussion by dmisdm** [quarto-dev/quarto-cli#1909] [Q&A] [answered]
Hey there!
Sorry if this is a redundant question, I just haven't yet worked out the answer after scrounging around!

Is rendering OJS output/viewofs (or even Julia) in PDFs supported? As far as I can tell, It's only doable when targeting HTML, since all I get when rendering to PDF is the code, without any ability for code folding either.

If I'm correct, and this feature isn't supported, I'd be delighted to try and implement this.
Could someone point me to the best starting place or provide links to relevant resources/source code?

Much obliged for any help 😇

This project is sick.

**Comment by rgaiacs** [comment]
Observable (OJS) cells are processed by the browser. The best way to export it to PDF is to print from your browser.

What content from Julia are you trying to export?

**Reply by dmisdm** [reply]
Appreciate the response!
Julia was just an example, since i noticed that cells weren't computing then rendering within PDFs. Personally, I'm more interested in getting OJS at the moment.
Since I'm looking for that "academic PDF" style, I don't think printing from a browser is viable unfortunately.

Do you by any chance know which part of the architecture handles executing code and rendering it?

Also, If i'm understanding correctly, it's not possible, and i should close this question?

**Comment by rgaiacs** [comment]
> Julia was just an example

I don't have Julia installed in my machine to provide you with a example. I believe that Quarto will try to render your document using Jupyter because it has a Julia cell and you are required to have a Julia kernel.

> Personally, I'm more interested in getting OJS at the moment.
> Since I'm looking for that "academic PDF" style, I don't think printing from a browser is viable unfortunately.

Can you share the reason that you want OJS? Would R work for you?

One of the advantages of OJS is it's graph resolution of cell order. The way that I had work around this with R is big portion of my data processing is in a separate R script that export the important variables to a `.rda` file which is loaded by default.

> Do you by any chance know which part of the architecture handles executing code and rendering it?

For OJS, it is a Javascript library. Quartos creates the HTML with the OJS cells and leave for the browser to render it. It is for math with TeX.

**Comment by cscheid** [comment, marked as answer]
Thanks for the kind words! You're correct that this feature is currently not supported, and we appreciate the offer and enthusiasm.

Unfortunately, adding support for HTML screen captures in PDF format is significantly trickier than it looks at first. There are a variety of reasons for this, but they all come down to the order in which operations currently happen in quarto. and by the time the PDF renderer figures out it needs to screenshot an HTML element, it's too late to schedule the rendering of the HTML format.

With that said, this is an important feature and it's 100% on our radar (we just had a design meeting about it yesterday in fact). The current expectation is that this should land anywhere from 3-9 months out.

**Reply by dmisdm** [reply]
Thanks for the response 😇
That makes a lot of sense! The change in architecture/pipeline for a relatively small feature doesn't feel like the juice is worth the squeeze.
I've started having a play around in the codebase, do you accept external PRs? (not to say I know what i'm doing yet lol). I do have an idea to get me by, but it ain't clean.
No pressure obviously. Happy to use a fork worst case!

Thanks again!

I'll mark this as answered for the time being!

**Reply by harrylojames** [reply]
+1 this would be really cool! @cscheid Is there an issue I can follow to track progress on this (I couldn't find one)?

**Reply by cscheid** [reply]
There isn't one. The plan is still in place but we've changed priorities to be able to land the 1.4 features we intend to, so my timeline is (clearly) no longer accurate and I can't give you a new one yet.

**Reply by frscott** [reply]
Just wondering if it was put back on any timeline or if it is sidelined indefinitely? Also what is the best workaround currently? Thanks!

**Reply by cscheid** [reply]
When there are updates we will post them.

**Reply by frscott** [reply]
This doesn't answer either of my questions. I get that this is open source and that you guys have bigger priorities. But you can still be kind in response to questions.

**Reply by cscheid** [reply]
@frscott We unfortunately have to balance the ~30 messages we get each day with the time budget we have, and your message doesn't get to the point where I have the time commitment to give a bigger answer. I get where you're coming from, but both of your message have no new content, have been answered before, and interrupt our work (because we can't know whether they have new content or a bug that we need to address).

You are reading unkindness where "time constraint" is the actual answer.

I literally answered "so my timeline is (clearly) no longer accurate and I can't give you a new one yet." above. Please understand that there actual hundreds (if not thousands) of you and only three of us working full-time as engineers on quarto-cli. If we take the time to answer each and every "updates?" question we have (like I am doing here), we would be have no time to actually work on the product. That's the unfortunate reality.

**Reply by frscott** [reply]
Makes sense! Thanks for responding!

_I literally answered "so my timeline is (clearly) no longer accurate and I can't give you a new one yet." above._

That was 1.5 years ago. I was just wondering if things had changed. Sorry!

**Reply by cscheid** [reply]
Here's a longer personal answer: https://bsky.app/profile/cscheid.net/post/3ls25wg6mnk2y

**Reply by frscott** [reply]
Thanks for letting me see your perspective. I am sorry I hit a sore spot with my response. When you say upvote a request does that just mean upvote this question? As you can tell I have very little experience with open source projects. In the future I will not ask about updates.

**Reply by mcanouil** [reply]
@frscott On discussion, it's the arrow you can see at the bottom of the "main" post(s):
<img width="73" alt="Button with an up arrow and the number two on the right" src="https://github.com/user-attachments/assets/e51d71d8-757b-4202-88ba-8d520395c4d1" />

At the discussion board level, you can find the same arrow (matching the one from the first original post of the thread).
<img width="909" alt="Screenshot of a GitHub discussion with the upvote arrow on the left showing the number 2" src="https://github.com/user-attachments/assets/908504f6-91dc-4e6b-9171-5cbb3e3dfab6" />

On issues,  use reactions such as thumb up to let the maintainers know you are interested without triggering notifications to ALL participants but still allowing maintainers to sort/filter the issue based on the number of reactions (they can't sort/filter with comments).
<img width="532" alt="Screenshot of the reaction button at the bottom of issues with the thumb up emoji being selected" src="https://github.com/user-attachments/assets/8204564e-b93c-4882-83aa-de5bf3b664fd" />

**Reply by frscott** [reply]
Thanks! I upvoted the post. I don't see this request as an issue yet is that correct?

**Reply by mcanouil** [reply]
as said, it's the reaction button.

**Comment by jjallaire** [comment]
We do accept PRs w/ a contributor agreement: https://quarto.org/about.html#contribute

**Reply by beyond2060** [reply]
I would love to be able to include Observable Plot graphs in PDFs. 

⭐️ Quarto totaly rocks! ⭐️
